src/lib/ams.c

```diff
@@ -30,6 +30,10 @@ DECLARE_SOF_UUID("ams", ams_uuid, 0xea9c4bca, 0x5b7d, 0x48c6,
 
 DECLARE_TR_CTX(ams_tr, SOF_UUID(ams_uuid), LOG_LEVEL_INFO);
 
+/* Key-phrase detected AMS message uuid: 80a11122-b36c-11ed-afa1-0242ac120002*/
+DECLARE_SOF_RT_UUID("ams_kpd_msg", ams_kpd_msg_uuid, 0x80a11122, 0xb36c, 0x11ed,
+		    0xaf, 0xa1, 0x02, 0x42, 0xac, 0x12, 0x00, 0x02);
+
```

Comment:
AMS should be a service that uses UUIDs blindly. Definitions of 3rd party module GUIDs should not be there.

Even if it was the case, this should be conditionally added on KPB being compiled in.

Reply:
@iganakov we do need a solution where the UUIDs are discovered at runtime, i.e. we have no hard coding in the AMS core. The AMS core should be usable by any module without AMS knowing about that module. This will mean extending the AMS API so that new clients can register their UUIDs (and any other meta data) at runtime (needs to support FW boot and post boot registrations).

Reply:
@plbossart @lgirdwood Sure, I agree that messages shouldn't be hardcoded in core AMS implementation. What I'd like to do is to put ams messages definitions for SOF built-in modules into one common place, e.g. `src\include\sof\lib\ams_msg.h`, thus it will be easier to manage them. 3rd party modules can define their own messages independently.

Reply:
I'm not sure that would make it easier to manage them. Both the consumer and the producer need access to the UUID and they'd also usually need access to a type declaration for the data, transferred with the AMS. So they'll have a common header. Then why don't they declare that UUID in that header too? That way when adding new AMS users they won't need to touch a global header with all UUIDs.